plugin/path_checkout_sets.go

```diff
@@ -316,23 +334,54 @@ func storeSet(ctx context.Context, storage logical.Storage, setName string, set
 	return storage.Put(ctx, entry)
 }
 
+func (b *backend) checkInNewServiceAccount(ctx context.Context, storage logical.Storage, serviceAccountName string) (isUserErr bool, err error) {
+	lock := locksutil.LockForKey(b.checkOutLocks, serviceAccountName)
+	lock.Lock()
+	defer lock.Unlock()
+
+	_, err = b.checkOutHandler.Status(ctx, storage, serviceAccountName)
+	if err == nil {
+		// We actually want to receive ErrNotFound here because that would indicate
+		// we're not already managing the potential to check this service account out
+		// through another set.
+		return true, fmt.Errorf("%s is already managed by another set, please remove it and try again", serviceAccountName)
+	}
+	if err != ErrNotFound {
+		// This is probably a persistent issue with reaching storage.
```

Comment:
Since we don't know what this is (only what it isn't), I'd avoid speculating on the cause.